shell/src/main/java/org/apache/accumulo/shell/commands/CreateTableCommand.java

```diff
@@ -150,29 +152,30 @@ public int execute(final String fullCommand, final CommandLine cl, final Shell s
     }
 
     // Copy configuration options if flag was set
+    Map<String,String> srcTableConfig = null;
     if (cl.hasOption(createTableOptCopyConfig.getOpt())) {
       String srcTable = cl.getOptionValue(createTableOptCopyConfig.getOpt());
       if (cl.hasOption(createTableOptExcludeParentProps.getLongOpt())) {
         // copy properties, excludes parent properties in configuration
-        Map<String,String> tableProps =
+        srcTableConfig =
             shellState.getAccumuloClient().tableOperations().getTableProperties(srcTable);
-        tableProps.entrySet().stream()
-            .filter(entry -> Property.isValidTablePropertyKey(entry.getKey()))
-            .forEach(entry -> initProperties.put(entry.getKey(), entry.getValue()));
       } else {
         // copy configuration (include parent properties)
-        final Map<String,String> configuration =
+        srcTableConfig =
             shellState.getAccumuloClient().tableOperations().getConfiguration(srcTable);
-        configuration.entrySet().stream()
-            .filter(entry -> Property.isValidTablePropertyKey(entry.getKey()))
-            .forEach(entry -> initProperties.put(entry.getKey(), entry.getValue()));
       }
+      srcTableConfig.entrySet().stream()
+          .filter(entry -> Property.isValidTablePropertyKey(entry.getKey()))
+          .forEach(entry -> initProperties.put(entry.getKey(), entry.getValue()));
     }
 
     // if no defaults selected, remove, even if copied from configuration or properties
     if (cl.hasOption(createTableNoDefaultIters.getOpt())) {
-      Set<String> initialProps = IteratorConfigUtil.generateInitialTableProperties(true).keySet();
+      // handles if default props were copied over
+      Set<String> initialProps = IteratorConfigUtil.getInitialTableProperties().keySet();
```

Comment:
keeping same functionality for now, but this is deleting the DefaultKeySizeConstraint when no default iterators is set, not sure if this is intentional